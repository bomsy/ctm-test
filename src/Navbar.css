@media all and (min-width: 320px) and (max-width : 480px) {
  .nav-bar {
    display: none;
  }
}

@media all and (min-width: 480px) and (max-width : 768px) {
  .nav-bar {
    display: none;
  }
}

@media all and (min-width: 768px) and (max-width: 1024px) {
  .nav-bar {
    display: block;
  }
  .nav-bar ul {
    padding: 0;
  }
  
  .nav-bar li {
    display: inline;
    padding: 25px 15px 15px 15px;
    border-radius: 10px;
    margin-right: 10px;
    background-image: -webkit-gradient(linear, 0% 0%, 0% 100%, from(#1a3c7f), to(#000034))
  }
  
  .nav-bar li.current {
    background: #04132a;
    padding-left: 25px;
  }
  
  .nav-bar li a {
    text-decoration: none;
    color: #ffffff;
  }
  
  .nav-bar li a span {
    display: none;
  }
}

@media all and (min-width: 1024px) {
  .nav-bar {
    display: block;
  }
  .nav-bar ul {
    padding: 0;
  }
  
  .nav-bar li {
    display: inline;
    padding: 25px 15px 15px 15px;
    border-radius: 10px;
    margin-right: 10px;
    background-image: -webkit-gradient(linear, 0% 0%, 0% 100%, from(#1a3c7f), to(#000034))
  }
  
  .nav-bar li.current {
    background: #04132a;
    padding-left: 25px;
  }
  
  .nav-bar li a {
    text-decoration: none;
    color: #ffffff;
  }
  
  .nav-bar li a span {
    position: relative;
    top: -5px;
  }
}



.nav-icon {
  content: '';
  display: inline-block;
  background: url('../assets/icons.svg');
  width: 25px;
  height: 21px;
  margin-right: 10px;
}

.home {
  background-position: 0 0;
}

.car {
  background-position: -70px 0;
}

.pet {
  background-position: -204px 0;
}

.finance {
  background-position: -302px 0;
  width: 28px;
}

.life {
  background-position: -170px 0;
}

.buisness {
  background-position: -436px 0;
}

.travel {
  background-position: -237px 0;
}