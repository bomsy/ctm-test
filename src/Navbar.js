"use strict";
import React, { Component } from "react";

export default class Navbar extends Component {
  constructor() {
    super();
  }

  render() {
    return (
      <nav>
        <a href="#" id="menu-icon"></a>
        <ul>
          <li>
            <a href="#" id="home-icon">
              <i className="nav-icon home"></i>
            </a>
          </li>
          <li>
            <a href="#">
              <i className="nav-icon car"></i>Vehicles
            </a>
          </li>
          <li>
            <a href="#">
              <i className="nav-icon pet"></i>Home & pet
            </a>
          </li>
          <li>
            <a href="#">
              <i className="nav-icon finance"></i>Finances
            </a>
          </li>
          <li>
            <a href="#" >
              <i className="nav-icon life"></i>Life
            </a>
          </li>
          <li>
            <a href="#" >
              <i className="nav-icon buisness"></i>Buisness
            </a>
          </li>
          <li>
            <a href="#" >
              <i className="nav-icon travel"></i>Travel
            </a>
          </li>
        </ul>
      </nav>
    );
  }
}
