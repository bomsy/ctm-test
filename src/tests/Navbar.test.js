import React from "react";
import Navbar from "../Navbar.js";
import renderer from "react-test-renderer";

test("Should load the navbar ", () => {
  const component = renderer.create(<Navbar />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
