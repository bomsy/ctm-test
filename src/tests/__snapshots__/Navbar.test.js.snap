// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should load the navbar  1`] = `
<nav
  className="nav-bar"
>
  <ul>
    <li
      className="current"
    >
      <a
        href="#"
        id="home-icon"
      >
        <i
          className="nav-icon home"
        />
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        <i
          className="nav-icon car"
        />
        <span>
          Vehicles
        </span>
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        <i
          className="nav-icon pet"
        />
        <span>
          Home & pet
        </span>
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        <i
          className="nav-icon finance"
        />
        <span>
          Finances
        </span>
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        <i
          className="nav-icon life"
        />
        <span>
          Life
        </span>
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        <i
          className="nav-icon buisness"
        />
        <span>
          Business
        </span>
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        <i
          className="nav-icon travel"
        />
        <span>
          Travel
        </span>
      </a>
    </li>
  </ul>
</nav>
`;
